pkg/api/grpcServer.go

```diff
@@ -133,8 +135,11 @@ func (s *grpcServer) dispatcher(stream grpc.Stream, methodName string,
 			worker(ctx2, requestBag, responseBag, request, response)
 
 			sendLock.Lock()
-			tracker.GetResponseAttributes(responseBag, responseAttrs)
-			err := s.sendMsg(stream, response)
+			if err = respTracker.ApplyBag(responseBag, 0, responseAttrs); err != nil {
```

Comment:
Shouldn't this be `respTracker.ApplyBag(responseBag, req.RequestIndex, responseAttrs)` instead of a hardcoded `0` for the context id?